From f81fab8170271e0fa7448af7b3396b23aa557704 Mon Sep 17 00:00:00 2001
From: "commit-queue@webkit.org"
 <commit-queue@webkit.org@268f45cc-cd09-0410-ab3c-d52691b4dbfc>
Date: Sun, 5 Feb 2017 07:02:30 +0000
Subject: [PATCH] Static Analyzer: Value stored to 'recordedMachineThreads'
 during its initialization is never read
 https://bugs.webkit.org/show_bug.cgi?id=167845

Patch by Joseph Pecoraro <pecoraro@apple.com> on 2017-02-04
Reviewed by Saam Barati.

Source/JavaScriptCore/heap/MachineStackMarker.cpp:151:14: warning: Value stored to 'recordedMachineThreads' during its initialization is never read
        auto recordedMachineThreads = m_set.take(machineThreads);
             ^~~~~~~~~~~~~~~~~~~~~~   ~~~~~~~~~~~~~~~~~~~~~~~~~~

* heap/MachineStackMarker.cpp:
(JSC::ActiveMachineThreadsManager::remove):

git-svn-id: http://svn.webkit.org/repository/webkit/trunk@211678 268f45cc-cd09-0410-ab3c-d52691b4dbfc
---
 Source/JavaScriptCore/heap/MachineStackMarker.cpp |  2 +-
 2 files changed, 15 insertions(+), 1 deletion(-)

diff --git a/Source/JavaScriptCore/heap/MachineStackMarker.cpp b/Source/JavaScriptCore/heap/MachineStackMarker.cpp
index df77939..e33f3b2 100644
--- a/Source/JavaScriptCore/heap/MachineStackMarker.cpp
+++ b/Source/JavaScriptCore/heap/MachineStackMarker.cpp
@@ -149,7 +149,7 @@ public:
     {
         LockHolder managerLock(m_lock);
         auto recordedMachineThreads = m_set.take(machineThreads);
-        RELEASE_ASSERT(recordedMachineThreads = machineThreads);
+        RELEASE_ASSERT(recordedMachineThreads == machineThreads);
     }
 
     bool contains(MachineThreads* machineThreads)
-- 
2.9.3

